### PR TITLE
fix: added all oidc-client configs to AuthProvider

### PR DESCRIPTION
**Pull request by mancioshell**
AuthProvider should support every config available on UserManager (#539)

### CODE REVIEW

**Comment by mancioshell** [comment]
@bjerkio

File: src/AuthContextInterface.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export interface AuthProviderSignOutProps {
   signoutRedirect?: boolean | unknown;
 }
 
-export interface AuthProviderProps {
+interface UserManagerProps {
```
Comment (cobraz) [inline src/AuthContextInterface.ts:26]:
```suggestion
export interface AuthProviderProps extends OidcClientSettings {
```

👆 not tested 😅

**Review by cobraz — APPROVED** [review]
Thank you for this contribution!

You open up an interesting feature for this library here. We would normally ask people to use their own UserManager for cases where the library didn't have the required properties.

I'll be needing to discuss this further before merging it, but the implementation seems good 👍

**Comment by cobraz** [comment]
@ryancole What do you think? Personally, I probably want to extend this into a new major version together with some other refactoring. This change alone isn't backwards-incompatible, but it changes how developers use `oidc-react`.

**Comment by pamapa** [comment]
Iff you are going this way keep attention to the now duplicated props: e.g. `oidc-client client_secret` vs `oidc-react clientSecret`.
Currently only `oidc-react clientSecret` is possible to pass in. When want to pass in all props from OidcClientSettings i strongly advice to drop the duplicated oidc-react props in favor of the oidc-client props.

On the other hand its already quite easy for the users to inject a customized UserManager.

The mapping can be seen here:
```
return new UserManager({
    authority,
    client_id: clientId,
    client_secret: clientSecret,
    redirect_uri: redirectUri,
    silent_redirect_uri: silentRedirectUri || redirectUri,
    post_logout_redirect_uri: postLogoutRedirectUri || redirectUri,
    response_type: responseType || 'code',
    scope: scope || 'openid',
    loadUserInfo: loadUserInfo != undefined ? loadUserInfo : true,
    popupWindowFeatures: popupWindowFeatures,
    popup_redirect_uri: popupRedirectUri,
    popupWindowTarget: popupWindowTarget,
    automaticSilentRenew,
  });
```

**Comment by cobraz** [comment]
> Iff you are going this way keep attention to the now duplicated props: e.g. `oidc-client client_secret` vs `oidc-react clientSecret`.
> Currently only `oidc-react clientSecret` is possible to pass in. When want to pass in all props from OidcClientSettings i strongly advice to drop the duplicated oidc-react props in favor of the oidc-client props.
> 
> On the other hand its already quite easy for the users to inject a customized UserManager.

Having thought about this PR for a while now, I realize we probably don't want all the potential arguments/configuration values from `oidc-client`. My reasoning behind it is this: As stated from @pamapa, you can quite easily inject a custom UserManager. However, what we should consider for `oidc-react` is to keep our implementation as easy/simple as possible, but let it be possible to customize. 

I am closing this pull request now, but there is nothing final about that. I would love to get some feedback for potential API changes like this, and the general usability of `oidc-react`. This is too a breaking change, which I think we should evaluate further with the community. :)

Thank you so much for your continued contribution!

**Pull request closed without merging**